src/main/java/net/minecraftforge/common/extensions/IForgeRawTagBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.extensions;
+
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.Tag;
+
+public interface IForgeRawTagBuilder
+{
+    default Tag.Builder getRawBuilder()
+    {
+        return (Tag.Builder)this;
+    }
+    
+    /**
+     * @apiNote internal, called when a raw builder is written to json to add forge additions (e.g. the remove list)
+     */
+    default void serializeTagAdditions(final JsonObject tagJson)
+    {
+        Tag.Builder rawBuilder = this.getRawBuilder();
+        List<Tag.BuilderEntry> removeEntries = rawBuilder.getRemoveEntries();
+        if (!removeEntries.isEmpty())
+        {
+            JsonArray removeEntriesAsJsonArray = new JsonArray();
+            for (Tag.BuilderEntry proxy : removeEntries)
+            {
+                proxy.getEntry().serializeTo(removeEntriesAsJsonArray);
+            }
+            tagJson.add("remove", removeEntriesAsJsonArray);
+        }
+    }
+    
+    /**
+     * Adds a tag proxy to the remove list to generate a json with
+     * @param proxy Tag proxy (either an item or another tag, usually)
+     * @return builder for chaining
+     * @apiNote internal; the other methods are preferable to call
+     */
+    default Tag.Builder removeProxy(final Tag.BuilderEntry proxy)
```

Comment:
All of these should just be called remove.
Just like all the add functions are named add.
The only changes are the conflicts in parameters.
So remove, removeElement, removeTag

Reply:
Fair enough, renamed the methods here as suggested. Also removed usages of apiNote per orion's recommendations.